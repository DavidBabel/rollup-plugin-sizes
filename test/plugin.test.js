"use strict";

var util = require("util"),
    
    rollup = require("rollup").rollup,
    
    plugin = require("../index.js");

describe("rollup-plugin-sizes", () => {
    let out;

    function log(...args) {
        out += util.format(...args) + "\n";
    }

    beforeEach(() => (out = ""));
    
    describe("single-entry bundle", () => {
        it("should show basic output", () =>
            rollup({
                input : "./test/specimens/entry.js",

                plugins : [
                    require("../index.js")({
                        log
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out).toMatchSnapshot())
        );

        it("should show detailed output ", () =>
            rollup({
                input : "./test/specimens/entry.js",

                plugins : [
                    require("../index.js")({
                        log,

                        details : true
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out).toMatchSnapshot())
        );
    });
});
