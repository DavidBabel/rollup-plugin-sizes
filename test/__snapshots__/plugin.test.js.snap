// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-sizes single-entry, multiple-output bundle should show basic output 1`] = `
"./test/specimens/1-to-many/a.js:
app - 65 B (100.00%)
"
`;

exports[`rollup-plugin-sizes single-entry, multiple-output bundle should show detailed output  1`] = `
"./test/specimens/1-to-many/a.js:
app - 65 B (100.00%)
	a.js - 44 B (67.69%)
	b.js - 21 B (32.31%)
"
`;

exports[`rollup-plugin-sizes single-entry, single-output bundle should show basic output 1`] = `
"./test/specimens/1-to-1/a.js:
app - 65 B (100.00%)
"
`;

exports[`rollup-plugin-sizes single-entry, single-output bundle should show detailed output  1`] = `
"./test/specimens/1-to-1/a.js:
app - 65 B (100.00%)
	a.js - 44 B (67.69%)
	b.js - 21 B (32.31%)
"
`;
